File: app/src/main/java/com/example/liput/ViewerActivity.java
package com.example.liput;

import android.os.Bundle;
import android.webkit.WebView;
import androidx.appcompat.app.AppCompatActivity;
import java.io.File;

public class ViewerActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        WebView webView = new WebView(this);
        setContentView(webView);
        String fileName = getIntent().getStringExtra("file");
        File file = new File(getFilesDir(), fileName);
        webView.loadUrl("file://" + file.getAbsolutePath());
    }
}

File: app/src/main/java/com/example/liput/ViewerActivity.kt
package com.example.liput

import android.os.Bundle
import android.webkit.WebView
import androidx.appcompat.app.AppCompatActivity
import java.io.File

class ViewerActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val webView = WebView(this)
        setContentView(webView)
        val fileName = intent.getStringExtra("file") ?: return
        val file = File(filesDir, fileName)
        webView.loadUrl("file://${file.absolutePath}")
    }
}
